pkg/registry/types.go

```diff
@@ -219,6 +230,15 @@ func (gd *GVKDependency) Validate() []error {
 	return errs
 }
 
+// Validate will validate GVK dependency type and return error(s)
+func (ld *LabelDependency) Validate() []error {
+	errs := []error{}
+	if ld.Label == "" {
```

Comment:
Why is this an error? IMO it could be invalid to omit the label field entirely from the dependency specification, but it should be OK to set it to any string, even the empty string.

Reply:
I'm fine with the suggestion. Will address this.

Reply:
Well, given this `label` field is a string. A empty string == there is no `label` field at all.

Reply:
If the struct field is the empty string, then either the label field was omitted or the label field was included and explicitly set to the empty string. It's impossible to distinguish between both cases here.

Reply:
That's what I meant. So either way, omitting the field or set the field to empty string is invalid from the validation standpoint. Plus, why should we allow an empty string as a label?

Reply:
> why should we allow an empty string as a label
My point is that disallowing the empty string is an arbitrary rule. Why is the empty string different from any other string? I always prefer to have fewer special cases as a rule.

I won't die on this hill, and I don't expect it to ruin anyone's day if they can't use the empty string, but this is an undocumented behavior without a test case.

Reply:
The zero value of a string is an empty string. That is a difference between empty string vs other strings in my perspective. You assign a string an empty string is equivalent to not assigning anything to that string at all. As you said before, there is impossible to distinguish between both cases. I think it is better to error out when they forget to assign a value to string or the value that is empty because why would anyone wants to do so in reality. We can revisit this issue if there is a demand to have an empty string as a value for a label. Plus, we will find out if someone asks for it when bundle validation will surface this up in the pipeline anyway.

Reply:
I don't think I've explained my problem with this clearly enough, but it's not very significant, so I'll resolve this thread.